src/workerd/api/sockets.c++

```diff
@@ -179,8 +181,11 @@ jsg::Ref<Socket> Socket::startTls(jsg::Lock& js, jsg::Optional<TlsOptions> tlsOp
     }
   }
   // All non-secure sockets should have a tlsStarter.
-  kj::Own<kj::AsyncIoStream> secure = KJ_ASSERT_NONNULL(*tlsStarter)(acceptedHostname);
-  return setupSocket(js, kj::mv(secure), kj::mv(options), kj::mv(tlsStarter), true, kj::mv(domain));
+  KJ_ASSERT_NONNULL(*tlsStarter)(acceptedHostname);
```

Comment:
(Comment on the code above this, which I can't comment directly on because it's not modified in this PR.)

You say we don't need to flush the WritableStream because it's backed by a KJ stream which doesn't do buffering. However, the WritableStream itself does buffering. Or rather, an application can perform multiple write()s in sequence without waiting for previous ones to complete, thus creating a queue. Should we flush it?

Reply:
Yes those need to be flushed

Reply:
In one of our previous chats you said:

> HOWEVER, WritableStreamSink CAN be buffered, if it's a compressed stream

The WritableStreamSink is backed by an AsyncIoStream which does no compression, so I assumed we don't need to worry about flushing. I'll ping you both in our discussion on this so we can go into more details.